Automattic/wp-calypso
# Theme Showcase: Add tabbed navigation

#### Changes proposed in this Pull Request

* Add tabbed navigation for "All" and "Recommended" themes to the Showcase.
* This takes advantage of existing filter behavior by setting a `filter` state and navigating to the filter's unique path `/themes/filter/[tag-slug]/[site]`.
* New tabs can be added by tagging themes in the Showcase with a specific tag, then pointing the new tab to that tag.

This is not 100% yet -- trying to go back and forth between tabs can cause the wrong themes to show, but I'm not sure why. 🤔 

**Before**

![screencapture-calypso-localhost-3000-themes-caroauditsimple-wordpress-com-2021-06-29-12_31_17](https://user-images.githubusercontent.com/2124984/123836828-21690b80-d8d8-11eb-8921-ad9c11a852c9.png)

**After**

<img width="1386" alt="Screen Shot 2021-06-29 at 12 26 40 PM" src="https://user-images.githubusercontent.com/2124984/123836881-32b21800-d8d8-11eb-8ff5-2d54d970ef61.png">


#### Testing instructions

* Switch to this PR
* Navigate to `/themes/site`
* Check out the tabs above the search bar! Click around and make sure they work as expected
* Check the showcase when logged out to make sure the tabs appear and work as expected
* Check the styles on mobile to make sure they work and nothing visually breaks

Related to #54083
